libfwupdplugin/fu-device.c

```diff
@@ -7512,44 +7512,44 @@ fu_device_add_instance_u32(FuDevice *self, const gchar *key, guint32 value)
 }
 
 /**
- * fu_device_build_instance_id:
+ * fu_device_build_instance_id_strv:
  * @self: a #FuDevice
  * @error: (nullable): optional return location for an error
  * @subsystem: (not nullable): subsystem, e.g. `NVME`
- * @...: pairs of string key values, ending with %NULL
+ * @keys: (nullable): %NULL terminated array of string keys
  *
- * Creates an instance ID from a prefix and some key values.
+ * Creates an instance ID from a prefix and an array of key values.
  * If the key value cannot be found, the parent and then proxy is also queried.
  *
  * If any of the key values remain unset then no instance ID is added.
  *
  *	fu_device_add_instance_str(dev, "VID", "1234");
  *	fu_device_add_instance_u16(dev, "PID", 5678);
- *	if (!fu_device_build_instance_id(dev, &error, "NVME", "VID", "PID", NULL))
+ *	const gchar *keys[] = {"VID", "PID", NULL};
+ *	if (!fu_device_build_instance_id_strv(dev, &error, "NVME", keys))
  *		g_warning("failed to add ID: %s", error->message);
  *
  * Returns: %TRUE if the instance ID was added.
  *
- * Since: 1.7.7
+ * Since: 2.0.14
  **/
 gboolean
-fu_device_build_instance_id(FuDevice *self, GError **error, const gchar *subsystem, ...)
+fu_device_build_instance_id_strv(FuDevice *self,
+				 GError **error,
+				 const gchar *subsystem,
+				 gchar **keys)
```

Comment:
ohh if we no longer have varargs, I think this should have GError at the end, e.g. self, subsystem, keys, error

Reply:
done